src/App.vue

```diff
@@ -1,13 +1,16 @@
 <template>
   <router-view v-slot="{ Component }">
     <transition name="pages" mode="out-in">
-      <component :is="Component" />
+      <with-auth>
+        <component :is="Component" />
+      </with-auth>
```

Comment:
I created a separate component to prevent unauthenticated user visit protected routes (routes that require login)